tools/snp.sh

```diff
@@ -904,13 +918,203 @@ attest_guest() {
+    dependencies_installed=1
+  fi
+}
+
+install_common_dependencies(){
+  # pip issue on 20.04 - some openssl bug
+  #sudo rm -f "/usr/lib/python3/dist-packages/OpenSSL/crypto.py"
+  pip install sev-snp-measure
+
+  # Rust is required to build snpguest
+  curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -sSf | sh -s -- -y
+
+  source "${HOME}/.cargo/env" 2>/dev/null
+}
+
+rhel_install_dependencies() {
+  local dependencies_installed_file="${WORKING_DIR}/dependencies_already_installed"
+  
+  check_if_dependencies_installed
+  # If dependencies already exist
+  if [[ $dependencies_installed -eq 1 ]]; then
+    return 0
+  fi
+  
+  # Step 1: Activate RedHat Subscription
+  rhel_subscription_mgr_set_login
+  sudo subscription-manager register --username ${RHEL_SUBS_MGR_USER} --password ${RHEL_SUBS_MGR_PASS} --force
+
+  # Build dependencies
+  sudo dnf install -y git
+  sudo dnf install -y make automake gcc gcc-c++ kernel-devel 
+
+  # Enable RedHat Repository for qemu dependencies
+  sudo subscription-manager register --username ${RHEL_SUBS_MGR_USER} --password ${RHEL_SUBS_MGR_PASS} --force
+  sudo subscription-manager repos --enable codeready-builder-for-rhel-9-x86_64-rpms
+
+  # qemu dependencies
+  sudo dnf install -y ninja-build
+  sudo dnf install -y pkg-config
+  sudo dnf install  -y glib2-devel
+  sudo dnf install  -y pixman-devel
+  sudo dnf install -y libslirp-devel
+   
+  # ovmf dependencies
+  sudo dnf install -y uuid-devel
+  sudo dnf install -y iasl
+  sudo dnf remove nasm
+  install_nasm_from_source
+
+  # kernel dependencies
+  sudo dnf install -y rsync
+  sudo dnf install -y ncurses-devel
+  
+  # libssl-dev is openssl-devel in RHEL
+  # rpm-build -- Scripts and executable programs used to build packages
+  sudo dnf install -y rpm-build
+
+  # cloud-utils dependency
+  sudo dnf install -y cloud-init
+  
+  # sev-snp-measure
+  sudo dnf install -y python3-pip
+
+  install_common_dependencies
+  echo "true" > "${dependencies_installed_file}"
+}
+
+fedora_install_dependencies() {
+  local dependencies_installed_file="${WORKING_DIR}/dependencies_already_installed"
+  
+  check_if_dependencies_installed
+  # If dependencies already exist
+  if [[ $dependencies_installed -eq 1 ]]; then
+    return 0
+  fi
+  
+  # Build dependencies
+  sudo dnf update
+  sudo dnf install -y git make ninja-build gcc glib2 glib2-devel pixman pixman-devel meson
+  
+  # For fix of FAILED: libqemuutil.a.p/util_async.c.o
+  # Uncomment wget command for applying patch d66ba6dc for async.c as per https://gitlab.com/qemu-project/qemu/-/issues/1655
+  # wget "https://raw.githubusercontent.com/qemu/qemu/d66ba6dc1cce914673bd8a89fca30a7715ea70d1/util/async.c" -O "${SETUP_WORKING_DIR}/AMDSEV/qemu/util/async.c"
+
+  sudo dnf install -y libuuid libuuid-devel
+  sudo dnf install -y python
+
+  # For fixing Failed to execute command make tbuild
+  sudo dnf remove nasm
+  install_nasm_from_source
+  sudo dnf install -y acpica-tools zstd rpm-build dwarves perl
```

Comment:
I feel it'd be better to split this up into multiple lines for the sake of uniformity